arch/arc/core/secureshield/secure_sys_services.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2018 Synopsys, Inc. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <errno.h>
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <zephyr/types.h>
+#include <soc.h>
+#include <toolchain.h>
+
+#include <arch/arc/v2/secureshield/arc_secure.h>
+
+#define IRQ_PRIO_MASK (0xffff << ARC_N_IRQ_START_LEVEL)
+/*
+ * @brief read secure auxiliary regs on behalf of normal mode
+ *
+ * @param aux_reg  address of aux reg
+ *
+ * Some aux regs require secure privilege, this function implements
+ * an secure service to access secure aux regs. Check should be done
+ * to decide whether the access is valid.
+ */
+static s32_t arc_s_aux_read(u32_t aux_reg)
+{
+	return -1;
+}
+
+/*
+ * @brief write secure auxiliary regs on behalf of normal mode
+ *
+ * @param aux_reg address of aux reg
+ * @param val, the val to write
+ *
+ *  Some aux regs require secure privilege, this function implements
+ * an secure service to access secure aux regs. Check should be done
+ * to decide whether the access is valid.
+ */
+static s32_t arc_s_aux_write(u32_t aux_reg, u32_t val)
+{
+	if (aux_reg == _ARC_V2_AUX_IRQ_ACT) {
+		/* 0 -> CONFIG_NUM_IRQ_PRIO_LEVELS allocted to secure world
+		 * left prio levels allocated to normal world
+		 */
+		val &= IRQ_PRIO_MASK;
+		_arc_v2_aux_reg_write(_ARC_V2_AUX_IRQ_ACT, val |
+		(_arc_v2_aux_reg_read(_ARC_V2_AUX_IRQ_ACT) & (~IRQ_PRIO_MASK)));
+
+		return  0;
+	}
+
+	return -1;
+}
+
+/*
+ * @brief allocate interrupt for normal world
+ *
+ * @param intno, the interrupt to be allocated to normal world
+ *
+ * By default, most interrupts are configured to be secure in initialization.
```

Comment:
I'd expect that Secure mode at system initialization defines certain interrupts as Normal, and rest as Secure. Not sure Normal mode needs this call.

Reply:
There may be some dynamic cases.  If the interrupt allocation is static, yes, no need such a call.